statistics/handle/handle.go

```diff
@@ -56,8 +56,10 @@ const (
 type statsCache struct {
 	tables map[int64]*statistics.Table
 	// version is the latest version of cache.
-	version  uint64
-	memUsage int64
+	version uint64
+	// minorVersion is to differentiate the cache when version is same.
+	minorVersion uint64
```

Comment:
It seems confused. If I update the stats item, which one should be increased? Are there any principles or rules we should obey?

Reply:
Could we explain how we use it here?

Reply:
Notice that for function `func (sc statsCache) update()`, a param `newVersion` is required. There are 3 callers of this function in total except extended stats component:
1) `LoadNeededHistograms` would insert column histograms into stats cache;
2) `GetPartitionStats` would insert a pseudo table stats;
3) the `Update()` procedure of stats;

After 1) and 2), the stats cache is actually at the same version as the one before the modification, so `newVersion` for `func (sc statsCache) update()` in these 2 cases are same as the old one, i.e, there is no version bump. Only the third case would bump the stats version if new stats are read from storage.

There would be lost of modifications if 1) and 2) happens concurrently(extended stats modification also belongs to this group then), that's why I introduce `minorVersion`, to differentiate stats caches which have trivial differences but have same version.

Reply:
> Are there any principles or rules we should obey?

We don't need to care about this `minorVersion` when writing code, if we use same `version` in `update()`, `minorVersion` would automatically be increased internally.

Reply:
After discussing it with @eurekaka offline, I finally got it... Please add more comments about this variable, especially a case.

Reply:
`(h *Handle) UpdateErrorRate` and `(h *Handle) UpdateStatsByLocalFeedback` also used `updateStatsCache()` without version bump, seems they also need modified?